lib/src/dart.dart

```diff
@@ -49,73 +43,20 @@ class _DirectiveCollector extends GeneralizingAstVisitor {
 }
 
-/// Runs [code] in an isolate.
-///
-/// [code] should be the contents of a Dart entrypoint. It may contain imports;
-/// they will be resolved in the same context as the host isolate. [message] is
-/// passed to the [main] method of the code being run; the caller is responsible
-/// for using this to establish communication with the isolate.
-///
-/// [packageRoot] controls the package root of the isolate. It may be either a
-/// [String] or a [Uri].
-///
-/// If [snapshot] is passed, the isolate will be loaded from that path if it
-/// exists. Otherwise, a snapshot of the isolate's code will be saved to that
-/// path once the isolate is loaded.
-Future runInIsolate(String code, message,
-    {packageRoot, String snapshot}) async {
-  if (snapshot != null && fileExists(snapshot)) {
-    log.fine("Spawning isolate from $snapshot.");
-    if (packageRoot != null) packageRoot = Uri.parse(packageRoot.toString());
-    try {
-      // Make the snapshot URI absolute to work around sdk#8440.
-      await Isolate.spawnUri(p.toUri(p.absolute(snapshot)), [], message,
-          packageRoot: packageRoot);
-      return;
-    } on IsolateSpawnException catch (error) {
-      log.fine("Couldn't load existing snapshot $snapshot:\n$error");
-      // Do nothing, we will regenerate the snapshot below.
-    }
-  }
-
-  await withTempDir((dir) async {
-    var dartPath = p.join(dir, 'runInIsolate.dart');
-    writeTextFile(dartPath, code, dontLogContents: true);
-    await Isolate.spawnUri(p.toUri(p.absolute(dartPath)), [], message,
-        packageRoot: packageRoot);
-
-    if (snapshot == null) return;
-
-    ensureDir(p.dirname(snapshot));
-    var snapshotArgs = <String>[];
-    if (packageRoot != null) snapshotArgs.add('--package-root=$packageRoot');
-    snapshotArgs.addAll(['--snapshot=$snapshot', dartPath]);
-    var result = await runProcess(Platform.executable, snapshotArgs);
-
-    if (result.success) return;
-
-    // Don't emit a fatal error here, since we don't want to crash the
-    // otherwise successful isolate load.
-    log.warning("Failed to compile a snapshot to "
-        "${p.relative(snapshot)}:\n" +
-        result.stderr.join("\n"));
-  });
-}
-
 /// Snapshots the Dart executable at [executableUrl] to a snapshot at
 /// [snapshotPath].
 ///
 /// If [packagesFile] is passed, it's used to resolve `package:` URIs in the
 /// executable. Otherwise, a `packages/` directory or a package spec is inferred
 /// from the executable's location.
 ///
-/// If [id] is passed, it's used to describe the executable in logs and error
-/// messages.
+/// If [packageName] and [pathInPackage] are passed, they are used to describe the
+/// executable in logs and error messages.
 Future snapshot(Uri executableUrl, String snapshotPath,
-    {Uri packagesFile, AssetId id}) async {
-  var name = log.bold(id == null
+    {Uri packagesFile, String packageName, String pathInPackage}) async {
```

Comment:
Consider replacing the `packageName` and `pathInPackage` parameters with an `name` parameter and make the caller responsible for deciding what that name could look like. Then the first line could just be `name ??= executableUrl.toString()`.

Reply:
Done